### PR TITLE
feat: enable high priority queries for the braze

### PR DESCRIPTION
**Pull request by egdbear**
This is the follow-up [Gravity PR](https://github.com/artsy/gravity/pull/16932) where we introduced high priority endpoints for the braze.

image

cc @artsy/diamond-devs

### CODE REVIEW

**Review by joeyAghion — APPROVED** [review]

File: src/schema/v2/me/myCollectionInfo.ts [hunk 1]
```diff
@@ -236,10 +236,12 @@ export const MyCollectionInfo: GraphQLFieldConfig<any, ResolverContext> = {
       return null
     }
 
-    context.userID = id
+    const userId = id || context.xImpersonateUserID
+
+    context.userID = userId
```
Comment (mzikherman) [inline src/schema/v2/me/myCollectionInfo.ts:241]: I know this line isn't 'new', but it seems sketchy to me...how come `context` needs to be set like this? It just jumps out as something with mutability and idk - seems weird

Comment (mzikherman) [reply src/schema/v2/me/myCollectionInfo.ts:241]: Yea, I don't think this is needed, the user id is _already_ in the context. This resolver shouldn't be taking `id` from the first argument, but from the context already. I can PR that update.

Comment (mzikherman) [reply src/schema/v2/me/myCollectionInfo.ts:241]: Fixed here - https://github.com/artsy/metaphysics/pull/5338

Comment (egdbear) [reply src/schema/v2/me/myCollectionInfo.ts:241]: Would it be all right to get `xImpersonateUserID` here in the resolver?

File: src/index.ts [hunk 2]
```diff
@@ -170,13 +170,15 @@ const graphqlServer = graphqlHTTP((req, res, params) => {
     defaultTimezone = timezone
   }
 
+  // Currently the xImpersonateUserID is only used for Braze purposes
+  const userID = xUserID || xImpersonateUserID
```
Comment (egdbear) [inline src/index.ts:174]: Maybe this makes more sense here @mzikherman ?

**Comment by egdbear** [comment]
All right, waiting for the green to merge:

However, we can't use the `similarToRecentlyViewedConnection` query yet because that query relies on the `recently_viewed_artwork_ids` field from the `/me` endpoint under me loader.

Nonetheless, this can be achieved by impersonating the `me/modules` endpoint from Gravity and calling it in the fallback of the `similarArtworksLoader` in the `SimilarToRecentlyViewed` query.

File: src/index.ts (hunk 2 above)
Comment (mzikherman) [reply src/index.ts:174]: This is a global-ish change, but I _think_ is ok. We usually use `X-User-Id` has the user id of the token making the request. It was out of convenience - the `X-Access-Token` is enough to use `meLoader` and get the id, but we use the id in a bunch of places and so passing it in as a header (both Force/Eigen do that) lets a resolver use it directly w/o needing to fetch `meLoader` first.

That all being said...I _think_ a fallback to the impersonate header makes sense.

**Pull request merged**